Feature/sprint11

ElanyFly opened the pull request:
Спринт 11
Подключение API

ElanyFly committed:
Сверстано 2 экрана заглушек для поиска
+ темная тема

ElanyFly committed:
имплементировала библиотеки ретрофита

ElanyFly committed:
интерфейс для получения списка треков.
подключила библиотеку OkHttp для логирования запросов и ответов.

ElanyFly committed:
метод getClient - ретрофит с возможностью логирования интернет запросов.
Вынесла текст ошибок в ресурсы Стринг.

ElanyFly committed:
inputText.setOnEditorActionListener test toast

ElanyFly committed:
получение запроса

ElanyFly committed:
видимость окон ошибки

ElanyFly committed:
конвертация миллисекунд в мм:сс

DmitrijEgorow commented:
- [ ] Белый текст на белом фоне невозможно прочитать
- [ ] Цвет курсора не соответствует дизайну

https://github.com/user-attachments/assets/1998de34-5e2d-4504-bf70-7ee32566364b

DmitrijEgorow commented:
Я бы убрал этот промежуток между поисковой строкой и элементом при пролистывании списка, но для первого элемента в списке его, конечно же, нужно сохранить

![Screenshot_20240731_004513](https://github.com/user-attachments/assets/8572c2eb-6eda-4e5d-a47c-4de05ea1ee3f)

DmitrijEgorow commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -38,12 +57,13 @@ class SearchActivity : AppCompatActivity() {
         }
 
-        val recyclerView = findViewById<RecyclerView>(R.id.recyclerView)
+
 
 
```
Много пустых строк

DmitrijEgorow commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -101,8 +173,35 @@ class SearchActivity : AppCompatActivity() {
+            addInterceptor(logging)
+        }
+
+        val newRetrofit = Retrofit.Builder()
```
Почему new?

ElanyFly replied:
Да никакой конкретной причины. Просто чтобы как-то её по имени отличать.

ElanyFly committed:
поправлен курсор окна поиска
исправлен цвет ввода текста в тёмной теме
часть атрибутов ЭдитТекста выведены в отдельный стиль.
